api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirObservationDaoImpl.java

```diff
@@ -175,4 +220,53 @@ protected Obs deproxyResult(Obs result) {
 		obs.setConcept(deproxyObject(obs.getConcept()));
 		return obs;
 	}
+	
+	int getMax(SearchParameterMap theParams) {
+		return ((NumberParam) theParams.getParameters(FhirConstants.MAX_SEARCH_HANDLER).get(0).getParam()).getValue()
+		        .intValue();
+		
+	}
+	
+	protected List<String> handleRanking(List<Object[]> list, int max) {
+		List<String> results = new ArrayList<>();
+		int cur_rank = 0;
+		
+		for (int var = 0; var < list.size() && cur_rank < max; var++) {
+			cur_rank++;
+			results.add((String) list.get(var)[0]);
+			
+			while (var + 1 < list.size() && list.get(var + 1)[2].equals(list.get(var)[2])) {
+				results.add((String) list.get(var + 1)[0]);
+				var++;
```

Comment:
1. To increase readability of this block, it's better if we can fetch `list.get(var+1)` and `list.get(var)` as separate meaningful variables beforehand. 
2. Shouldn't we compare the `Date` of the successive observations rather than the `Object`?

Reply:
I had typecast it to `Date` but apparently, it was a redundant typecast. But since we'll be storing it as a separate `Date` type variables now, this situation won't arise